### ISSUE TITLE
GCStress: JitStress: Crash in _dbggcreport

### ISSUE THREAD

**Issue by swaroop-sridhar** [dotnet/coreclr#2965]
The failure repros most of the time under the settings:

```
COMPlus_GCStress=0xC
COMPlus_JitStress=2
COMPlus_ZapDisable=1
```

```
corerun JIT/Methodical/refany/_dbggcreport/_dbggcreport.exe

Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at JitTest.StressTest.Scenario2()
   at JitTest.StressTest.Main()
/home/swaroops/bin/runn: line 3: 32364 Aborted                 (core dumped) ./corerun ~/test/$1.exe
134
```

**Comment by sivarv** [comment]
Mkreany test case. @LLITCHEV - could be fixed by your PR changes.  Please verify and close.

**Comment by LLITCHEV** [comment]
Seems to be fixed by PR #2860
